ethereum/client.go

```diff
@@ -210,6 +210,87 @@ func toBlockNumArg(number *big.Int) string {
 	return hexutil.EncodeBig(number)
 }
 
+// Transaction returns the transaction response of the Transaction identified
+// by *RosettaTypes.TransactionIdentifier hash
+func (ec *Client) Transaction(
+	ctx context.Context,
+	blockIdentifier *RosettaTypes.BlockIdentifier,
+	transactionIdentifier *RosettaTypes.TransactionIdentifier,
+) (*RosettaTypes.Transaction, error) {
+	if transactionIdentifier.Hash != "" {
+		var raw json.RawMessage
+		err := ec.c.CallContext(ctx, &raw, "eth_getTransactionByHash", transactionIdentifier.Hash)
+		if err != nil {
+			return nil, fmt.Errorf("%w: transaction fetch failed", err)
+		} else if len(raw) == 0 {
+			return nil, ethereum.NotFound
+		}
+
+		// Decode transaction
+		var body rpcTransaction
+
+		if err := json.Unmarshal(raw, &body); err != nil {
+			return nil, err
+		}
+
+		var header *types.Header
+		if blockIdentifier.Hash != "" {
+			header, err = ec.blockHeaderByHash(ctx, blockIdentifier.Hash)
+		} else {
+			header, err = ec.blockHeaderByNumber(ctx, big.NewInt(blockIdentifier.Index))
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("%w: could not get block header for %x", err, blockIdentifier.Hash)
+		}
+
+		receipt, err := ec.transactionReceipt(ctx, body.tx.Hash())
+		if receipt.BlockHash != *body.BlockHash {
+			return nil, fmt.Errorf(
+				"%w: expected block hash %s for transaction but got %s",
+				ErrBlockOrphaned,
+				body.BlockHash.Hex(),
+				receipt.BlockHash.Hex(),
+			)
+		}
+		if err != nil {
+			return nil, fmt.Errorf("%w: could not get receipt for %x", err, body.tx.Hash())
+		}
+
+		var traces *Call
+		var rawTraces json.RawMessage
+		var addTraces bool
+		if header.Number.Int64() != GenesisBlockIndex { // not possible to get traces at genesis
+			addTraces = true
+			traces, rawTraces, err = ec.getTransactionTraces(ctx, body.tx.Hash())
+			if err != nil {
+				return nil, fmt.Errorf("%w: could not get traces for %x", err, body.tx.Hash())
+			}
+		}
+
+		gasUsedBig := new(big.Int).SetUint64(receipt.GasUsed)
+		feeAmount := gasUsedBig.Mul(gasUsedBig, body.tx.GasPrice())
```

Comment:
Are you planning to rebase our EIP 15519 change?

Reply:
yes. I will do that as a separate PR since this is already a big PR.